### PR TITLE
Adding slxos_vlan module

### PR DESCRIPTION
**Pull request by bigmstone**
##### SUMMARY
This PR adds the 'slxos_vlan' module.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
`slxos_vlan`

##### ANSIBLE VERSION
```
⟩ ansible --version
ansible 2.6.0 (slxos-vlan b7d2cc541b) last updated 2018/04/23 16:11:55 (GMT -500)
  config file = None
  configured module search path = [u'/Users/mstone/dev/stackstorm/ansible/library']
  ansible python module location = /Users/mstone/dev/stackstorm/ansible/lib/ansible
  executable location = /Users/mstone/dev/stackstorm/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 29 2018, 15:58:25) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]

```

##### ADDITIONAL INFORMATION

```
⟩ ansible-playbook --ask-pass -i hosts.yml vlan.yml
SSH password:
ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to have been in '/Users/mstone/dev/stackstorm/ansible-slxos/vlan.yml': line 7, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: Set Vlan
      ^ here
```
```
⟩ ansible-playbook --ask-pass -i hosts.yml vlan.yml
SSH password:

PLAY [dc2leaf2] ****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Set Vlan] ****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
ok: [dc2leaf2]

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
dc2leaf2                   : ok=1    changed=0    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by bigmstone** [comment]
@gundalow This PR is ready for merge.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@bigmstone Can you rebase please, thanks in advance

File: lib/ansible/modules/network/slxos/slxos_vlan.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+      - ID of the VLAN. Range 1-4094.
+    required: true
+  interfaces:
+    description:
+      - List of interfaces that should be associated to the VLAN.
+    required: true
+  delay:
+    description:
+      - Delay the play should wait to check for declarative intent params values.
+    default: 10
```
Comment (ganeshrn) [inline lib/ansible/modules/network/slxos/slxos_vlan.py:54]: type should be `int`

**Review by ganeshrn — CHANGES_REQUESTED** [review]

**Review by ganeshrn — APPROVED** [review]
lgtm, requires a minor doc change

**Comment by bigmstone** [comment]
@ganeshrn docs change taken care of
@gundalow conflict resolved

File: lib/ansible/modules/network/slxos/slxos_vlan.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/modules/network/slxos/slxos_vlan.py:54]: The only type we support in DOCUMENTATION is `type: bool` which automatically adds choices,

This line needs removing.

type int should be specified in argspec

File: lib/ansible/modules/network/slxos/slxos_vlan.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+      - ID of the VLAN. Range 1-4094.
+    required: true
+  interfaces:
+    description:
+      - List of interfaces that should be associated to the VLAN.
+    required: true
+  delay:
+    description:
+      - Delay the play should wait to check for declarative intent params values.
+    type: int
```
Comment (gundalow) [inline lib/ansible/modules/network/slxos/slxos_vlan.py]: Please remove this line.
The only `type` that is supported in `DOCUMENTATION` is `type: bool`, which is to automatically add the choices..

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
Hi, 
See if http://docs.ansible.com/ansible/devel/dev_guide/index.html helps with the rebase, thanks in advance.

**Review by gundalow — APPROVED** [review]

**Pull request merged**